BSPConversionCmd/Program.cs

```diff
@@ -22,39 +22,62 @@ class Options
 			[Option("prefix", Required = false, HelpText = "Prefix for the converted BSP's file name.")]
 			public string Prefix { get; set; }
 
-			[Value(0, MetaName = "input file", HelpText = "Input Quake 3 BSP/PK3 to be converted.", Required = true)]
-			public string InputFile { get; set; }
+            [Value(0, MetaName = "input file", HelpText = "Input Quake 3 BSP/PK3 file and its path to be converted. (i.e. C:\\path\\to\\file.pk3)", Required = true)]
+            public string InputFile { get; set; }
 
-			[Value(1, MetaName = "output directory", HelpText = "Output game directory for converted BSP/materials.", Required = true)]
-			public string OutputDirectory { get; set; }
-		}
+            [Value(1, MetaName = "output directory", HelpText = "Output game directory for converted BSP/materials. (i.e. C:\\path\\to\\output\\folder)", Required = false)]
+            public string OutputDirectory { get; set; }
+        }
 
 		static void Main(string[] args)
 		{
-			//args = new string[3];
-			//args[0] = @"c:\users\tyler\documents\tools\source engine\bspconvert\kopo_fe.pk3";
-			//args[1] = @"c:\users\tyler\documents\tools\source engine\bspconvert\output";
-			//args[2] = "--newbsp";
-
-			Parser.Default.ParseArguments<Options>(args)
-				.WithParsed(options =>
-				{
-					if (options.DisplacementPower < 2 || options.DisplacementPower > 4)
-						throw new ArgumentOutOfRangeException("Displacement power must be between 2 and 4.");
-
-					var converterOptions = new BSPConverterOptions()
-					{
-						noPak = options.NoPak,
-						skyFix = options.SkyFix,
-						DisplacementPower = options.DisplacementPower,
-						newBSP = options.NewBSP,
-						prefix = options.Prefix,
-						inputFile = options.InputFile,
-						outputDir = options.OutputDirectory
-					};
-					var converter = new BSPConverter(converterOptions, new ConsoleLogger());
-					converter.Convert();
-				});
-		}
+            //args = new string[3];
+            //args[0] = @"c:\users\tyler\documents\tools\source engine\bspconvert\kopo_fe.pk3";
+            //args[1] = @"c:\users\tyler\documents\tools\source engine\bspconvert\output";
+            //args[2] = "--newbsp";
+
+            string[] inputFilePaths = new string[] { };
+            List<string> inputEntries = new List<string>();
```

Comment:
I'd rather let the user specify a directory to convert instead of a list of files since it's a bit easier to set up and automate. Also I'm not sure if this will integrate well with the command line parser library that I'm using. Ideally it should automatically handle all of the parsing logic for you using the attributes from the `Options` class. https://github.com/commandlineparser/commandline

Reply:
Oh, interesting. I didn't consider dragging multiple files at once. I think that is possible to achieve with the command line parser. It should look something like this (I haven't tested this):
`[Option('i', "input", Required = true, HelpText = "Input file(s) to process", Separator = ',')]`